spinn_front_end_common/utilities/helpful_functions.py

```diff
@@ -303,6 +304,19 @@ def convert_vertices_to_core_subset(vertices, placements):
     return core_subsets
 
 
+def find_executable_start_type(machine_vertex, graph_mapper=None):
+    has_binary = isinstance(machine_vertex, AbstractHasAssociatedBinary)
+
+    if not has_binary:
+        return None
+    elif graph_mapper is not None:
+        app_vertex = graph_mapper.get_application_vertex(machine_vertex)
+        if isinstance(app_vertex, AbstractHasAssociatedBinary):
+            return app_vertex.get_binary_start_type()
+    else:
```

Comment:
Else not needed if you fix the return above

Reply:
actually. having looked closer. it seems this function isnt called anywhere.  so removing

Reply:
actually. its used by the bitfield compressor report. and the appliction chips used. so it is used.

Reply:
having relooked. i think i see what you were trying to say. please check new version